stablehlo/dialect/TypeInference.cpp

```diff
@@ -65,6 +65,8 @@ namespace hlo {
 // Utils for shape functions.
 //===----------------------------------------------------------------------===//
 
+static constexpr int64_t kDynamicSizePrintValue = -1;
```

Comment:
In the PR description, can you explain the rationale behind these changes? Also, why does it matter to type inference how dynamic sizes are printed? Aren't they printed as question marks anyway?